### DISCUSSION TITLE
React admin - Sugestões

### DISCUSSION THREAD

**Discussion by Juuhz** [frontendbr/forum#1379] [Discussão]
Galera, bom dia, tudo bem ?

Eu criei uma plataforma ( ainda não está online ), e quero pedir a ajuda de vocês, vejam:

## Introdução:
Atualmente toda a galera do universo react ( React Js / Native ) tem um problema que é: Editar seus textos / imagens de forma dinâmica, exemplo: 

- Se você tem um texto na sua home ou app e precisa muda-lo, terá que alterá-lo no código e depois buildar o projeto e rodar o deploy em produção. 

Isso acaba sendo custoso, ainda mais para quem faz sites para clientes onde tem bastante alterações de conteúdo ( blogs ou sites institucionais, por exemplo ).

Pensando nisso eu desenvolvi uma plataforma online onde você cria um dashboard totalmente personalizado. Podendo adicionar vários menus laterais, exemplo:

- Home
- Artigos
- Quem somos
- Contato
- etc... 

E dentro desses menus, você tem um CRUD onde cadastra seus textos, e no final do form de cadastro/edição, é gerado um código para você colocar em sua aplicação front-end, exemplo:

`<ReactContent token="hkjawkhjdlawhdlkajlakn" />`

OBS: esse componente será um pacote do NPM.

Essa linha puxará o conteúdo registrado online, de forma que você poderá alterá-la quando quiser.

-------

Atualmente existe uma solução chamada: React-admin:
https://marmelab.com/react-admin/Tutorial.html

Mas eu julgo ela muito complexa, você gastará um tempo só para fazer o painel rodar e depois terá que configurar tudo dentro dele.

-------

Queria pedir a opinião/ajuda de vocês nas seguintes perguntas.

1) Você acha minha ideia boa? Acha que irá agregar na comunidade ou é apenas bullshit?
2) Usaria em seus projetos?
3) Você utiliza ou conheçe alguma solução parecida ? 
4) React-admin seria um nome perfeito, mas já existe, oque você sugeriria como nome ?
5) Sugestões ou críticas construtivas ? 

Valeu galera, amo vocês <3

Atenciosamente,
Reinaldo Amorim.

**Comment by carloshenriqueribeiro** [comment]
Quando preciso fazer isso, costumo usar o BD. Usando o sistema de autenticação e autorização já implementado pela propria aplicação. Não acha mais simples ?

**Comment by carloshenriqueribeiro** [comment]
Considerações: 
- só token não será suficiente para organizar o permicionamento e acesso;
Registrada online exatamente onde ?

Blog engine ja não resolvem isso bem ?

**Comment by felipemfp** [comment]
Parece muito o que é possível fazer com [Gatsby usando uma fonte de conteúdo externa (CMS)](https://www.gatsbyjs.org/docs/headless-cms/).

Sobre textos de tela, geralmente usando qualquer framework i18n com fonte externa, você consegue mudar os textos sem precisar alterar diretamente no código.
